controllers/openstackcluster_controller.go

```diff
@@ -564,6 +564,15 @@ func bastionToOpenStackServerSpec(openStackCluster *infrav1.OpenStackCluster) *i
 	}
 	openStackServerSpec := openStackMachineSpecToOpenStackServerSpec(bastion.Spec, openStackCluster.Spec.IdentityRef, compute.InstanceTags(bastion.Spec, openStackCluster), az, nil, getBastionSecurityGroupID(openStackCluster), openStackCluster.Status.Network.ID)
 
+	if server != nil {
```

Comment:
This is the ugly part that I would agree to change/refactor.
Basically if an OpenStackServer already existed (the case for the bastion, which can be created then updated with a new spec), we want to compare Apples with Apples, so we need to get the Resolved & Resources fields too if they were here before. Otherwise it'll reconcile in infinite and the bastion will be deleted/recreated/deleted, etc.